linera-base/src/crypto.rs

```diff
@@ -396,6 +393,20 @@ where
     }
 }
 
+// Temporary struct to extend `Keccak256` with `io::Write`.
+struct Keccak256Ext(Keccak256);
```

Comment:
Is this used in places other than `CryptoHash::new`? Otherwise we might as well just call `update` directly there.

Reply:
We hash in Signature::new which uses interface from Write trait.

Reply:
Actually, even without that it'd require pretty big rewrite:
```rust
impl<'de, T, Hasher> Hashable<Hasher> for T
where
    T: BcsHashable<'de>,
    Hasher: io::Write,
{
    fn write(&self, hasher: &mut Hasher) {
        let name = <Self as HasTypeName>::type_name();
        // Note: This assumes that names never contain the separator `::`.
        write!(hasher, "{}::", name).expect("Hasher should not fail");
        bcs::serialize_into(hasher, &self).expect("Message serialization should not fail");
    }
}
```
write now we serialize `T` using the `Hashable` impl above. If I were to remove the wrapper (for which there exists a `io::Write` implementation then the `fn write` code above would have to be replicated in `CryptoHash::new` itself